sklearn/tests/test_random_projection.py

```diff
@@ -271,15 +304,24 @@ def test_random_projection_embedding_quality():
         assert 1 - eps < distances_ratio.min()
 
 
-def test_SparseRandomProj_output_representation():
+@pytest.mark.parametrize("coo_container", COO_CONTAINERS)
+@pytest.mark.parametrize("csr_container", CSR_CONTAINERS)
+def test_SparseRandomProj_output_representation(coo_container, csr_container):
+    data = make_sparse_random_data(
+        coo_container,
+        n_samples,
+        n_features,
+        n_nonzeros,
+        sparse_format=False,
+    )
     for SparseRandomProj in all_SparseRandomProjection:
         # when using sparse input, the projected data can be forced to be a
         # dense numpy array
         rp = SparseRandomProj(n_components=10, dense_output=True, random_state=0)
         rp.fit(data)
         assert isinstance(rp.transform(data), np.ndarray)
 
-        sparse_data = sp.csr_matrix(data)
+        sparse_data = csr_container(data)
```

Comment:
Thanks for the suggestion, I did it. I have used `global_random_seed` here, because I believe that `assert isinstance(...)` or` sp.issparse `would pass even if the data was different?
Should I change that to use  `random_state=0` ?

Reply:
Yes let's keep the `random_state` to 0. We can revisit this test in a latter PR to see how much flaky it is by changing the `random_state`.

Reply:
Thank you, @glemaitre, I have changed it to use `random_state=0`.